### PR TITLE
[Feature:CourseMaterials] Edit Course Materials Links

### LINKED ISSUE #5934: Edit link url for course material link
**What problem are you trying to solve with Submitty**
If the instructor typos the url, when uploading a course material link item,
currently they have to delete the item and re-add the item.

Let's let the edit the link url.

### PR DESCRIPTION
**Pull request by cjreed121**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Once you create a new link on the course materials page there is no way to edit it. The only effective way of editing it is to delete the link and create a new one.
Closes #5934

### What is the new behavior?
If you click the edit button on a link you are given additional fields to edit the title or url of the link.

### CODE REVIEW

File: site/public/js/server.js [hunk 1]
```diff
@@ -306,6 +306,20 @@ function newEditCourseMaterialsForm(path, dir, this_file_section, this_hide_from
         $("#all-sections-showing-yes", form).prop('checked',false);
         $("#all-sections-showing-no", form).prop('checked',true);
     }
+    if (is_link === "1") {
```
Comment (HenrikLam) [inline site/public/js/server.js:309]: Is there a reason to use `"1"` over `"true"`? Would `"true"` be more readable?

**Comment by HenrikLam** [comment]
If you create 2 links of different names, then change 1 of the link to have the same link, the link that was change seems to disappear (even if the link url is different). The old link also seems to linger and the user can no longer use that name within that directory despite it not being displayed anymore. 
image

File: site/public/js/server.js (hunk 1 above)
Comment (cjreed121) [reply site/public/js/server.js:309]: That would be because the value is actually "1" and not "true".

**Review by HenrikLam — COMMENTED** [review]
Works well, but the bug where if you have a slash as first character for directory name still exists.

**Review by HenrikLam — APPROVED** [review]
Seems to work well

**Pull request merged**